source/delegate/contracts/Delegate.sol

```diff
@@ -167,6 +169,14 @@ contract Delegate is IDelegate, Ownable {
       amountToStake,
       bytes32(uint256(address(this)) << 96) //NOTE: this will pad 0's to the right
     );
+
+    if (oldStakingAmount > amountToStake) {
+      // return excess stake back
+      require(
+        IERC20(indexer.stakingToken())
+        .transfer(msg.sender, oldStakingAmount - amountToStake), "STAKING_TRANSFER_FAILED"
+      );
+    }
```

Comment:
we have the conditions for:
```
if (oldStakingAmount < amountToStake)
if (oldStakingAmount > amountToStake)
```
but not when `oldStakingAmount == amountToStake` just to confirm, if it's equal we're assuming they're likely updating the rule and we probably don't need to reset the intent.

Reply:
yes

Reply:
could we forego setting the intent then at the end?

Reply:
so i'd prefer not having the delegate reach in and check if an index exists and whether an intent exists. I'll put in a check to forego at zero but any further checks to the internals of index doesn't see right.